api/cmd/main.go

```diff
@@ -194,6 +195,7 @@ func main() {
 		DB:                        db,
 		Enforcer:                  authEnforcer,
 		FeastCoreClient:           coreClient,
+		MlFlowClientFactory:       mlflowClientFactory,
```

Comment:
hmm i think `ClientFactory` is not really necessary here? Do you think it's cleaner if this is just `MLflowClient` field which accepts an object that satisfies `mlflow.Client` interface?

Reply:
See line 213, they call it MlpURL so its kinda inconsistent. But if you follow the pattern it seems that outside names seem to be initial caps and small characters for the rest of the acronym.

Reply:
now i know what you mean, let me change it.

Reply:
I'm not sure how to get away with not using a factory pattern in order to keep recreating a new client. I need to be able to mock the mlflow client in order to write unit tests.

We cant instantiate similar to how the core feast client does it as we are unable to keep the same state, i.e. mlflowTrackingUrl always changing per request. A singleton pattern would not suffice and a prototype pattern is required for this.